proving-system/stark/src/lib.rs

```diff
@@ -657,3 +657,18 @@ mod test_utils {
         ret
     }
 }
+
+#[derive(Debug, thiserror::Error)]
+pub enum ArithmeticError {
+    #[error("Integer overflow on an operation with {a} and {b}")]
+    IntegerOverflow { a: u64, b: u64 },
+}
+
+fn add(a: u64, b: u64) -> Result<u64> {
+    a.checked_add(b)
+        .ok_or(ArithmeticError::IntegerOverflow { a, b })
+}
+
+type Result<T, E = ArithmeticError> = std::result::Result<T, E>;
```

Comment:
This is great for testing, but we shouldn't merge this.

Let's discuss how to do this.